esphome/components/esp32/__init__.py

```diff
@@ -518,6 +519,20 @@ def _detect_variant(value):
     return value
 
 
+def _check_advanced(config):
+    if (
+        config.get(CONF_FRAMEWORK, {})
+        .get(CONF_ADVANCED, {})
+        .get(CONF_EXECUTE_FROM_PSRAM)
+        and config[CONF_VARIANT] != VARIANT_ESP32S3
+    ):
+        raise cv.Invalid(
+            f"'{CONF_EXECUTE_FROM_PSRAM}' is only supported on {VARIANT_ESP32S3} variant",
+            path=[CONF_ADVANCED, CONF_EXECUTE_FROM_PSRAM],
```

Comment:
Missing the start of the path =)
```suggestion
            path=[CONF_FRAMEWORK, CONF_ADVANCED, CONF_EXECUTE_FROM_PSRAM],
```